Private/macOS/notarize.sh

```diff
@@ -0,0 +1,159 @@
+#!/bin/bash
+
+# For detailed explanation see: https://developer.apple.com/documentation/security/notarizing_your_app_before_distribution/customizing_the_notarization_workflow
+
+usage() {
+    cat <<EOM
+
+$(basename $0) - Handy script to notarize a kernel extension file (KEXT)
+
+Usage: $(basename $0) -id <apple_id> -p <password> -k <path_to_kext>
+
+        -id or --appleid         # A valid Apple ID email address, account must have correct certificates available
+        -p  or --password        # The password for the specified Apple ID or Apple One-Time password (to avoid 2FA)
+        -k  or --kext            # The path to the kernel extension .kext file
+
+EOM
+    exit 0
+}
+
+declare arg_AppleId=""
+declare arg_Password=""
+declare arg_KextPath=""
+
+[ $# -eq 0 ] && { usage; }
+
+function parseArgs() {
+    arg_Positional=()
+    while [[ $# -gt 0 ]]; do
+        cmd="$1"
+        case $cmd in
+        --help | -h)
+            usage
+            shift
+            exit 0
+            ;;
+        --appleid | -id)
+            arg_AppleId=$2
+            shift
+            ;;
+        --password | -p)
+            arg_Password=$2
+            shift
+            ;;
+        --kext | -k)
+            arg_KextPath=$2
+            shift
+            ;;
+        *)
+            arg_Positional+=("$1")
+            shift
+            ;;
+        esac
+    done
+}
+
+parseArgs "$@"
+
+if [[ -z $arg_AppleId ]]; then
+    echo "[ERROR] Must supply valid / non-empty Apple ID!"
+    exit 1
+fi
+
+if [[ -z $arg_Password ]]; then
+    echo "[ERROR] Must supply valid / non-empty password!"
+    exit 1
+fi
+
+if [[ -z $arg_KextPath ]]; then
+    echo "[ERROR] Must supply valid / non-empty path to KEXT to notarize!"
+    exit 1
+fi
+
+declare bundle_id=`/usr/libexec/PlistBuddy -c "Print :CFBundleIdentifier" ${arg_KextPath}/Contents/Info.plist`
+
+if [[ -z $bundle_id ]]; then
+    echo "[ERROR] No CFBundleIdentifier found in KEXT Info.plist!"
+    exit 1
+fi
+
+echo "Notarizating $arg_KextPath"
+declare kext_zip=$arg_KextPath.zip
```

Comment:
nit: I understand that what I suggested is unnecessary, but using `${var}` looks better to me when concatenating something to it
```suggestion
declare kext_zip="${arg_KextPath}.zip"
```